src/main/java/games/chess/ChessGameState.java

```diff
@@ -0,0 +1,414 @@
+package games.chess;
+
+import core.AbstractGameState;
+import core.AbstractParameters;
+import core.actions.AbstractAction;
+import core.components.Component;
+import games.chess.components.ChessBoard;
+import games.GameType;
+import games.chess.actions.MovePiece;
+import games.chess.components.ChessPiece;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class ChessGameState extends AbstractGameState {
+
+    ChessBoard board = new ChessBoard();
+    //List of white pieces
+    List<ChessPiece> whitePieces = new ArrayList<>();
+    //List of black pieces
+    List<ChessPiece> blackPieces = new ArrayList<>();
+    //Game state counts 
+    Map<Integer, Integer> gameStateCounts = new HashMap<>();
+
+    //Number of moves without a pawn move or capture
+    int halfMoveClock = 0;
+
+
+    
+    
+
+    public ChessGameState(AbstractParameters gameParameters, int nPlayers) {
+        super(gameParameters, nPlayers);
+
+    }
+
+    @Override
+    protected GameType _getGameType() {
+        return GameType.Chess;
+    }
+
+    @Override
+    protected List<Component> _getAllComponents() {
+        List<Component> components = new ArrayList<>();
+        components.add(board);
+        return components;
+    }
+
+    @Override
+    protected ChessGameState _copy(int playerId) {
+        ChessGameState copy = new ChessGameState(getGameParameters(), getNPlayers());
+        copy.whitePieces = new ArrayList<>();
+        copy.whitePieces.addAll(whitePieces.stream().map(ChessPiece::copy).toList());
```

Comment:
You could replace this I think with copy.whitePieces = new ArrayList(whitePieces); which is much more concise and avoids streams

Reply:
Yes - my mistake...I forgot ChessPiece is mutable.